gammapy/datasets/simulate.py

```diff
@@ -26,6 +27,66 @@ class MapDatasetEventSampler:
     def __init__(self, random_state="random-seed"):
         self.random_state = get_random_state(random_state)
 
+    def _sample_coord_time_energy(self, dataset, temporal_model, t_delta="1 sec"):
```

Comment:
Is a fixed `t_delta` the simplest approach? One could take an oversampling factor as well. It could take the mean of the dataset duration and the model time interval

Reply:
I used the `t_delta` parameter so to be in line with the approach of `_sample_coord_time`. If we agree to follow your suggestion, we should change it in both functions for consistency. For now I implement it in the `_sample_coord_time_energy`.